### PR TITLE
Allowed some empty dirname in emulateObjectDirectories() method

### PR DESCRIPTION
**Pull request by ewake**
https://www.php.net/manual/en/function.empty.php#103756

### CODE REVIEW

File: src/Util.php [hunk 1]
```diff
@@ -291,13 +291,13 @@ protected static function emulateObjectDirectories(array $object, array $directo
             $listedDirectories[] = $object['path'];
         }
 
-        if (empty($object['dirname'])) {
+        if (empty($object['dirname']) && !is_numeric($object['dirname'])) {
```
Comment (GrahamCampbell) [inline src/Util.php]: What's wrong with a numeric directory name?

Comment (GrahamCampbell) [reply src/Util.php]: Oh, I see. Maybe just change the whole thing to `if (isset($object['dirname']) && $object['dirname'] !== null) {`. Is much clearer what the intent is then.

**Comment by ewake** [comment]
Hi there, i encountered problems when `$object['dirname']` is `'0'`, and I used this solution https://www.php.net/manual/en/function.empty.php#103756 .
I change it with your advice if you think it is better..

**Comment by frankdejonge** [comment]
Hi @ewake, thank you for your contribution. The assertion you're making check if for a known thing to exclude, rather than checking if the value is "wrong". Could you update it if it's set and if it's not an empty string. That's basically what we want to guard against. I can do it myself to, but if you like to contribute that's really welcome as well 👍

**Comment by ewake** [comment]
> Hi @ewake, thank you for your contribution. The assertion you're making check if for a known thing to exclude, rather than checking if the value is "wrong". Could you update it if it's set and if it's not an empty string. That's basically what we want to guard against. I can do it myself to, but if you like to contribute that's really welcome as well

Hi Frank, I updated the pull request and I hope I understood what you mean..
Otherwise my contribution will be limited to reporting the problem, that I hope someone can solve better than me :wink:

**Comment by frankdejonge** [comment]
Released as 1.0.59

**Pull request merged**